### DISCUSSION TITLE
Auto discover path based testcase

### DISCUSSION THREAD

**Discussion by Xuanwo** [risinglightdb/risinglight#338] [Ideas]
How about discovering test cases by path instead of hand writing every test case?

**Comment by skyzh** [comment]
That would be a good idea! But there are two problems to consider.

* Currently, sqllogictest is running as a Rust test case, and each file corresponds to one test case. If we write some auto-discovery logic, test cases name won't be show in `cargo test`.
* sqllogictest has an `include` syntax support, which enables test cases to reuse sql snippets. So we cannot simply scan all `slt` files and run them one by one. Some of them might be snippet for other files.

**Comment by xxchan** [comment]
resolved in https://github.com/risinglightdb/risinglight/pull/376
btw, I found GitHub disscussions don't support autolinking now. not very cool 🐶 https://github.com/github/feedback/discussions/2971
